### DISCUSSION TITLE
Multi-language support

### DISCUSSION THREAD

**Discussion by jsmorabito** [athensresearch/athens#789] [Feature Requests]
Inspired by Obsidian https://github.com/obsidianmd/obsidian-translations I'd like to open the discussion for adding an option to the settings menu to allow the user to select the language of menus, buttons, prompts, etc in Athens. Once a system is set up we can coordinate a process much like Obsidians to allow for translation support by contributors.

**Comment by Em-AK** [comment]
### Option 1
I have experience with https://github.com/ptaoussanis/tempura library. I would say it is the simple and accessible approach, with complete flexibility as a dictionary key can map to either a string `"Hello"`, a template string `"Hello %1"` or an arbitrary function (to handle plurals aor region specific grammar for example).
Tongue library https://github.com/tonsky/tongue follows the same approach (I have no experience with this one). Additionally Tongue handles numbers and dates formating according to the locale.
Both are clj/cljs compatible.

### Option 2
I read about some other i18n/l10n standards like [CLDR](http://cldr.unicode.org/) which seems somehow linked to [ICU message formats](https://unicode-org.github.io/icu/userguide/format_parse/messages/). I have not any experience with these, but it seems a bit overkill to me as I have no PhD in software internationalization/localization.

EDIT: I found https://github.com/thedavidmeister/cljs-i18n but it seems more related to numbers/dates/currency/... localization (formatting **and** parsing).

---
I like option 1 more.
EDIT: And I think it is best to keep translation and numbers/dates/ ... localization (format and parse) as independent (orthogonal) topics for now. Also I have no knowledge about what are the parsing capabilities of athens for now (regarding dates for example).

The roadmap to translating athens can be:
1. Load the chosen lib.
2. Extract each string to an `en` dictionary (a clojure map, or an .edn file) and set the "current locale" to a static `en`.
3. Recommend new features to add strings to the `en` dictionary, and start translating it to other locales.
4. Add a language selector in the settings, so that `translate` takes the "current locale" into account. And release this feature once a few languages are complete.

Open question: does this effort technically intersects with RTL support? #935

**Reply by Em-AK** [reply]
Now I realize that at least datetimes should also be localizable.
The parts highlighted in **green** are straightforward to localize:
* Created and Modified columns is just about rendering (I guess).
* the generated Current Time is just stored as text, so times generated before and after switching locales will not be with the same formatting
-> **is it a problem?**

The parts in **orange** involve a bit more work:
* thankfully daily notes can be identified with their `uid`
* switching locales would have the effect of updating the title of all daily notes to the new locale format, the new titles being derived from the `uid`s.

![image12](https://user-images.githubusercontent.com/1518393/116807822-a680b000-ab35-11eb-9eb8-2504b90a0a95.png)
![i18n-slash-commands](https://user-images.githubusercontent.com/1518393/116808066-28bda400-ab37-11eb-91b6-21b4088879ab.png)
![2021-05-02-113235_748x108_scrot](https://user-images.githubusercontent.com/1518393/116808747-d8484580-ab3a-11eb-8b33-430b268d45fa.png)
